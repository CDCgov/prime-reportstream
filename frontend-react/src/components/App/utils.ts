/**
 * Type-safe way to ensure refs are initialized only once.
 */
export function getService<T>(
    ref: React.MutableRefObject<T>,
    initFn: () => Exclude<T, null>,
) {
    if (ref.current !== null) {
        return ref.current;
    }
    const service = initFn();
    ref.current = service;
    return service;
}
