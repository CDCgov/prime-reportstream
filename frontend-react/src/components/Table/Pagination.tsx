import classnames from "classnames";
import {
    Button,
    IconNavigateBefore,
    IconNavigateNext,
} from "@trussworks/react-uswds";

export const OVERFLOW_INDICATOR = Symbol("…");
export type SlotItem = number | typeof OVERFLOW_INDICATOR;

const PaginationOverflow: React.FC = () => (
    <li
        className="usa-pagination__item usa-pagination__overflow"
        role="presentation"
    >
        <span>…</span>
    </li>
);

interface PaginationPageNumberProps {
    pageNum: number;
    setSelectedPage: (pageNum: number) => void;
    isCurrentPage: boolean;
    isLastPage: boolean;
}

const PaginationPageNumber: React.FC<PaginationPageNumberProps> = ({
    pageNum,
    setSelectedPage,
    isCurrentPage,
    isLastPage,
}) => {
    return (
        <li className="usa-pagination__item usa-pagination__page-no">
            <button
                {...(isCurrentPage && { "aria-current": "page" })}
                aria-label={`${isLastPage ? "last page, " : ""}Page ${pageNum}`}
                className={classnames(
                    ["usa-pagination__button", "rs-pagination-no-button"],
                    { "usa-current": isCurrentPage }
                )}
                onClick={() => setSelectedPage(pageNum)}
            >
                {pageNum}
            </button>
        </li>
    );
};

interface PaginationArrowProps {
    pageNum: number;
    setSelectedPage: (pageNum: number) => void;
    direction: "previous" | "next";
}

const PaginationArrow: React.FC<PaginationArrowProps> = ({
    pageNum,
    setSelectedPage,
    direction,
}) => {
    const isNext = direction === "next";
    const isPrevious = direction === "previous";
    const label = isPrevious ? "Previous" : "Next";
    const buttonClassName = classnames("rs-pagination-arrow-button", {
        "usa-pagination__previous-page": isPrevious,
        "usa-pagination__next-page": isNext,
    });
    return (
        <li className="usa-pagination__item usa-pagination__arrow">
            <Button
                aria-label={`${label} page`}
                className={buttonClassName}
                onClick={() => setSelectedPage(pageNum)}
                type="button"
                unstyled
            >
                {direction === "previous" && <IconNavigateBefore />}
                <span className="usa-pagination__link-text">{label}</span>
                {direction === "next" && <IconNavigateNext />}
            </Button>
        </li>
    );
};

export interface PaginationProps {
    slots: SlotItem[];
    setSelectedPage: (pageNum: number) => void;
    currentPageNum: number;
    label?: string;
}

const Pagination: React.FC<PaginationProps> = ({
    slots,
    setSelectedPage,
    currentPageNum,
    label = "Pagination",
}) => {
    const previousPageNum =
        currentPageNum - 1 > 0 ? currentPageNum - 1 : undefined;
    const lastSlot = slots[slots.length - 1];
    const nextPageNum =
        lastSlot === OVERFLOW_INDICATOR || lastSlot > currentPageNum
            ? currentPageNum + 1
            : undefined;

    return (
        <nav aria-label={label} className="usa-pagination">
            <ul className="usa-pagination__list">
                {previousPageNum && (
                    <PaginationArrow
                        pageNum={previousPageNum}
                        setSelectedPage={setSelectedPage}
                        direction="previous"
                    />
                )}
                {slots.map((s, i) => {
                    const key = String(s) + i;
                    if (s === OVERFLOW_INDICATOR) {
                        return <PaginationOverflow key={key} />;
                    }
                    return (
                        <PaginationPageNumber
                            key={key}
                            pageNum={s}
                            setSelectedPage={setSelectedPage}
                            isCurrentPage={s === currentPageNum}
                            isLastPage={i === slots.length - 1}
                        />
                    );
                })}
                {nextPageNum && (
                    <PaginationArrow
                        pageNum={nextPageNum}
                        setSelectedPage={setSelectedPage}
                        direction="next"
                    />
                )}
            </ul>
        </nav>
    );
};

export default Pagination;
