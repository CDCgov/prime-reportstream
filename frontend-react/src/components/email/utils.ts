export function oktaProperty(name: string) {
    return `$\{${name}}`;
}
