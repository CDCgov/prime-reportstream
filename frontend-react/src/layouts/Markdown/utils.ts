import { mergeWith } from "lodash";
import { AppConfig } from "../../config";

/**
 * Deep merge default meta with frontmatter meta. Has customizer that takes empty strings into account.
 */
export function createMeta(config: AppConfig, frontmatter: Frontmatter) {
    return mergeWith(
        {},
        config.PAGE_META.defaults,
        frontmatter.meta,
        (a, b) => {
            if (typeof b === "string" && b === "") return a;
            return undefined;
        },
    );
}
