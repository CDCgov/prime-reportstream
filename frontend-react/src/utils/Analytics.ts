import { getAppInsights } from "../TelemetryService";

export enum EventName {
    TABLE_FILTER = "Table Filter",
    SESSION_DURATION = "Session Duration",
    TABLE_PAGINATION = "Table Pagination",
}

export const trackAppInsightEvent = (eventName: string, eventData: any) => {
    const appInsights = getAppInsights();
    const telemetryEvent = {
        name: eventName,
        properties: eventData,
    };

    if (eventName !== "") appInsights?.trackEvent(telemetryEvent);
};
