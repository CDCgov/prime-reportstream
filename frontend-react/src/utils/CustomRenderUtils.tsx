import { ReactElement, ReactNode, Suspense } from "react";
import {
    render,
    RenderOptions,
    renderHook as renderHookOrig,
    RenderHookOptions,
    Queries,
    queries,
} from "@testing-library/react";
import {
    createMemoryRouter,
    Outlet,
    RouterProvider,
    RouteObject,
} from "react-router-dom";
import { QueryClientProvider } from "@tanstack/react-query";
import { HelmetProvider } from "react-helmet-async";
import { Fixture, MockResolver } from "@rest-hooks/test";
import { CacheProvider } from "rest-hooks";

import { SessionProviderBase } from "../contexts/Session";
import { AuthorizedFetchProvider } from "../contexts/AuthorizedFetch";
import { getTestQueryClient } from "../network/QueryClients";
import { FeatureFlagProvider } from "../contexts/FeatureFlag";
import { appRoutes } from "../AppRouter";
import AppInsightsContextProvider from "../contexts/AppInsights";
import config from "../config";

interface AppWrapperProps {
    children: ReactNode;
}

interface AppWrapperOptions {
    initialRouteEntries?: string[];
    restHookFixtures?: Fixture[];
}

function TestLayout() {
    return <Outlet />;
}

function createTestRoutes(
    routes: RouteObject[],
    element: ReactNode,
): RouteObject[] {
    return routes.map((r) => ({
        ...r,
        lazy: undefined,
        element: r.path !== "/" ? element : <TestLayout />,
        children: r.children
            ? createTestRoutes(r.children, element)
            : undefined,
    })) as RouteObject[];
}
export const AppWrapper = ({
    initialRouteEntries,
    restHookFixtures,
}: AppWrapperOptions = {}) => {
    // FUTURE_TODO: Replace children with <AppRouter /> if initialRouteEntries after mocking okta users
    // in tests is made easier for better coverage as we'd be able to test through
    // any custom route wrappers.
    // FUTURE_TODO: Remove MockResolver and restHookFixtures when removing react-hooks.
    return ({ children }: AppWrapperProps) => {
        /**
         * Dynamically makes the supplied children the return element for all
         * routes.
         * FUTURE_TODO: Remove this once okta user/session mocking is easier
         * and use <AppRouter /> instead.
         */
        const router = createMemoryRouter(
            createTestRoutes(appRoutes, children),
            {
                initialEntries: initialRouteEntries,
            },
        );
        return (
            <Suspense>
                <CacheProvider>
                    <HelmetProvider>
                        <AppInsightsContextProvider>
                            <SessionProviderBase
                                oktaAuth={{} as any}
                                authState={{}}
                                config={config}
                            >
                                <QueryClientProvider
                                    client={getTestQueryClient()}
                                >
                                    <AuthorizedFetchProvider
                                        initializedOverride={true}
                                    >
                                        <FeatureFlagProvider>
                                            {restHookFixtures ? (
                                                <MockResolver
                                                    fixtures={restHookFixtures}
                                                >
                                                    <RouterProvider
                                                        router={router}
                                                    />
                                                </MockResolver>
                                            ) : (
                                                <RouterProvider
                                                    router={router}
                                                />
                                            )}
                                        </FeatureFlagProvider>
                                    </AuthorizedFetchProvider>
                                </QueryClientProvider>
                            </SessionProviderBase>
                        </AppInsightsContextProvider>
                    </HelmetProvider>
                </CacheProvider>
            </Suspense>
        );
    };
};

interface RenderAppOptions extends RenderOptions, AppWrapperOptions {}

export const renderApp = (
    ui: ReactElement,
    {
        initialRouteEntries,
        restHookFixtures,
        ...options
    }: Omit<RenderAppOptions, "wrapper"> = {},
) => {
    return render(ui, {
        wrapper: AppWrapper({ initialRouteEntries, restHookFixtures }),
        ...options,
    });
};

export function renderHook<
    Result,
    Props,
    Q extends Queries = typeof queries,
    Container extends Element | DocumentFragment = HTMLElement,
    BaseElement extends Element | DocumentFragment = Container,
>(
    render: (initialProps: Props) => Result,
    options?: RenderHookOptions<Props, Q, Container, BaseElement>,
) {
    return renderHookOrig<Result, Props, Q, Container, BaseElement>(render, {
        wrapper: AppWrapper(),
        ...options,
    });
}

export { screen } from "@testing-library/react";
