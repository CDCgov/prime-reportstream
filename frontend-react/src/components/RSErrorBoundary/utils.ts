import { ErrorBoundaryProps } from "react-error-boundary";
import { RSConsole } from "../../utils/rsConsole/rsConsole";
import { isRSNetworkError } from "../../utils/RSNetworkError";

/**
 * Error handler that emits through rsConsole if available.
 */
export function createErrorLogger(
    rsConsole?: RSConsole,
): Exclude<ErrorBoundaryProps["onError"], undefined> {
    return (exception, info) => {
        if (!isRSNetworkError(exception)) {
            console.warn(
                "Please work to migrate all non RSError throws to use an RSError object.",
            );
        }
        if (rsConsole) {
            // React will always console.error all errors, regardless of boundary,
            // so just emit the telemetry.
            rsConsole._error(
                {
                    args: [exception, info.componentStack],
                    location: window.location.href,
                },
                rsConsole.severityLevels.error,
            );
        }
    };
}
