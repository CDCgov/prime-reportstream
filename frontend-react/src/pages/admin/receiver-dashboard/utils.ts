import { addMinutes, eachDayOfInterval, interval } from "date-fns";
import { RSReceiverStatus } from "../../../hooks/api/UseReceiversConnectionStatus/UseReceiversConnectionStatus";
import { DatePair } from "../../../utils/DateTimeUtils";

export enum SuccessRate {
    UNDEFINED = "UNDEFINED",
    ALL_SUCCESSFUL = "ALL_SUCCESSFUL",
    ALL_FAILURE = "ALL_FAILURE",
    MIXED_SUCCESS = "MIXED_SUCCESS",
}

export const SUCCESS_RATE_CLASSNAME_MAP = {
    [SuccessRate.UNDEFINED]: "success-undefined",
    [SuccessRate.ALL_SUCCESSFUL]: "success-all",
    [SuccessRate.ALL_FAILURE]: "failure-all",
    [SuccessRate.MIXED_SUCCESS]: "success-mixed",
};

export enum MatchingFilter {
    NO_FILTER,
    FILTER_NOT_MATCHED,
    FILTER_IS_MATCHED,
}

export const MATCHING_FILTER_CLASSNAME_MAP = {
    [MatchingFilter.NO_FILTER]: "",
    [MatchingFilter.FILTER_NOT_MATCHED]: "success-result-hidden",
    [MatchingFilter.FILTER_IS_MATCHED]: "",
};

/**
 * build the dictionary with a special path+key
 * @param dataIn
 */
export const sortStatusData = <
    T extends RSReceiverStatus[] | RSReceiverStatusParsed[],
>(
    dataIn: T,
): T => {
    const data = structuredClone(dataIn);
    const { orgNameMaxLength, receiverNameMaxLength } = data.reduce(
        (prev, curr) => {
            prev.orgNameMaxLength =
                curr.organizationName.length > prev.orgNameMaxLength
                    ? curr.organizationName.length
                    : prev.orgNameMaxLength;
            prev.receiverNameMaxLength =
                curr.receiverName.length > prev.receiverNameMaxLength
                    ? curr.receiverName.length
                    : prev.receiverNameMaxLength;

            return prev;
        },
        {
            orgNameMaxLength: 0,
            receiverNameMaxLength: 0,
        },
    );

    // sorting by organizationName, then receiverName, then connectionCheckStartedAt
    data.sort(
        (
            d1: RSReceiverStatus | RSReceiverStatusParsed,
            d2: RSReceiverStatus | RSReceiverStatusParsed,
        ) => {
            // ideally the shape of this type will change so that all receivers, regardless if any status data was found, will get sorted
            const [sortStrA, sortStrB] = [d1, d2].map(
                (x) =>
                    `${x.organizationName.padEnd(orgNameMaxLength, "-")}${x.receiverName.padEnd(receiverNameMaxLength, "-")}${new Date(x.connectionCheckStartedAt || Date.now()).toISOString()}`,
            );

            const result =
                sortStrA > sortStrB ? 1 : sortStrA < sortStrB ? -1 : 0;

            return result;
        },
    );

    return data;
};

export interface TimePeriodData {
    start: Date;
    end: Date;
    success: number;
    fail: number;
    isResultFilterMatch: boolean;
    successRateType: SuccessRate;
    matchingFilter: MatchingFilter;
    entries: RSReceiverStatusParsed[];
}

export function createStatusTimePeriodData({
    data,
    range: [startDate, endDate],
    filterResultMessage = "",
    timePeriodMinutes = 2 * 60,
}: {
    data: RSReceiverStatus[];
    range: DatePair;
    filterResultMessage?: string;
    timePeriodMinutes?: number;
}): ReceiverStatusTimePeriod[] {
    const inter = interval(startDate, endDate, { assertPositive: true });
    const numTimePeriodsPerDay = (24 * 60) / timePeriodMinutes;
    if (numTimePeriodsPerDay % 1 !== 0)
        throw new Error("Invalid time period duration");

    const timePeriodLabels = Array.from(Array(numTimePeriodsPerDay).keys()).map(
        (_, i) =>
            addMinutes(
                startDate,
                (i + 1) * timePeriodMinutes,
            ).toLocaleTimeString(),
    );
    const sortedData = sortStatusData(data).map((d) => ({
        ...d,
        connectionCheckCompletedAt: new Date(d.connectionCheckCompletedAt),
        connectionCheckStartedAt: new Date(d.connectionCheckStartedAt),
    }));
    const receiverIds = Array.from(
        new Set(sortedData.map((d) => d.receiverId)),
    );

    return receiverIds.map((id) => {
        const entries = sortedData.filter((d) => d.receiverId === id);
        const { organizationName, receiverName } = entries[0];
        const days = eachDayOfInterval(inter).map((day) => {
            const dayEntries = entries.filter(
                (e) =>
                    e.connectionCheckCompletedAt.toLocaleDateString() ===
                    day.toLocaleDateString(),
            );
            const dayString = day.toLocaleDateString();
            const timePeriods = timePeriodLabels.map((time) => {
                const start = new Date(`${day.toLocaleDateString()}, ${time}`);
                const end = addMinutes(start, timePeriodMinutes);

                const timePeriodEntries = dayEntries.filter((e) => {
                    return (
                        e.connectionCheckCompletedAt >= start &&
                        e.connectionCheckCompletedAt < end
                    );
                });
                const agg = timePeriodEntries.reduce(
                    (
                        agg,
                        { connectionCheckSuccessful, connectionCheckResult },
                    ) => {
                        if (connectionCheckSuccessful) agg.success += 1;
                        else {
                            agg.fail += 1;
                            if (
                                connectionCheckResult
                                    .toLowerCase()
                                    .includes(filterResultMessage.toLowerCase())
                            ) {
                                agg.isResultFilterMatch = true;
                            }
                        }
                        return agg;
                    },
                    {
                        success: 0,
                        fail: 0,
                        isResultFilterMatch: false as boolean,
                    },
                );

                const successRateType =
                    agg.success && agg.fail
                        ? SuccessRate.MIXED_SUCCESS
                        : !agg.success && !agg.fail
                          ? SuccessRate.UNDEFINED
                          : agg.success
                            ? SuccessRate.ALL_SUCCESSFUL
                            : SuccessRate.ALL_FAILURE;
                const matchingFilter = !filterResultMessage
                    ? MatchingFilter.NO_FILTER
                    : filterResultMessage && agg.isResultFilterMatch
                      ? MatchingFilter.FILTER_IS_MATCHED
                      : MatchingFilter.FILTER_NOT_MATCHED;

                return {
                    start: start,
                    end: end,
                    ...agg,
                    successRateType,
                    matchingFilter,
                    entries: timePeriodEntries,
                };
            });
            return {
                day,
                dayString,
                timePeriods,
                entries: dayEntries,
            };
        });

        const { success, fail } = days.reduce(
            (agg, curr) => {
                curr.timePeriods.forEach((c) => {
                    agg.fail += c.fail;
                    agg.success += c.success;
                });
                return agg;
            },
            { success: 0, fail: 0 },
        );
        const successRate =
            !!success || !!fail
                ? Math.round((100 * success) / (success + fail))
                : 0;
        const successRateType =
            !success && !fail
                ? SuccessRate.UNDEFINED
                : successRate === 0
                  ? SuccessRate.ALL_FAILURE
                  : successRate === 100
                    ? SuccessRate.ALL_SUCCESSFUL
                    : SuccessRate.MIXED_SUCCESS;

        return {
            organizationName,
            receiverName,
            id,
            successRate,
            successRateType,
            days,
        };
    });
}

export interface ReceiverStatusTimePeriod {
    organizationName: string;
    receiverName: string;
    id: number;
    successRate: number;
    successRateType: SuccessRate;
    days: {
        day: Date;
        dayString: string;
        timePeriods: {
            start: Date;
            end: Date;
            success: number;
            fail: number;
            isResultFilterMatch: boolean;
            successRateType: SuccessRate;
            matchingFilter: MatchingFilter;
            entries: RSReceiverStatusParsed[];
        }[];
        entries: RSReceiverStatusParsed[];
    }[];
}

export type RSReceiverStatusParsed = Omit<
    RSReceiverStatus,
    "connectionCheckStartedAt" | "connectionCheckCompletedAt"
> & {
    connectionCheckStartedAt: Date;
    connectionCheckCompletedAt: Date;
};

export function filterStatuses(
    statuses: ReceiverStatusTimePeriod[],
    receiverNameLike?: string,
    successRateLike?: SuccessRate,
) {
    return statuses.filter(({ receiverName, successRateType }) => {
        const result =
            (!receiverNameLike ||
                receiverName.toLowerCase().includes(receiverNameLike)) &&
            (!successRateLike ||
                successRateLike === SuccessRate.UNDEFINED ||
                successRateType === successRateLike);

        return result;
    });
}
