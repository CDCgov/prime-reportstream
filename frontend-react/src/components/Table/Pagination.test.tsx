import { fireEvent, screen, within } from "@testing-library/react";

import { renderWithBase } from "../../utils/CustomRenderUtils";

import Pagination, {
    PaginationProps,
    SlotItem,
    OVERFLOW_INDICATOR,
} from "./Pagination";

describe("Pagination", () => {
    test.each([
        {
            description: "on the first page of an unbounded set",
            slots: [1, 2, 3, 4, 5, 6, OVERFLOW_INDICATOR],
            currentPageNum: 1,
            expectedItems: ["1", "2", "3", "4", "5", "6", "…", "Next"],
        },
        {
            description: "on the second page of an unbounded set",
            slots: [1, 2, 3, 4, 5, 6, OVERFLOW_INDICATOR],
            currentPageNum: 2,
            expectedItems: [
                "Previous",
                "1",
                "2",
                "3",
                "4",
                "5",
                "6",
                "…",
                "Next",
            ],
        },
        {
            description: "on the first page of a bounded set",
            slots: [1, 2],
            currentPageNum: 1,
            expectedItems: ["1", "2", "Next"],
        },
        {
            description: "on the last page of a bounded set",
            slots: [1, 2],
            currentPageNum: 2,
            expectedItems: ["Previous", "1", "2"],
        },
        {
            description: "when there is only one page",
            slots: [1],
            currentPageNum: 1,
            expectedItems: ["1"],
        },
    ])(
        "Handles Previous and Next links $description",
        ({ slots, currentPageNum, expectedItems }) => {
            const props: PaginationProps = {
                slots: slots as SlotItem[],
                currentPageNum,
                setSelectedPage: jest.fn(),
            };
            renderWithBase(<Pagination {...props} />);

            const list = screen.getByRole("list");
            const { getAllByRole } = within(list);
            const items = getAllByRole("listitem");
            for (const [i, item] of items.entries()) {
                expect(item).toHaveTextContent(expectedItems[i]);
            }
        }
    );

    test("Clicking on pagination items invokes the setSelectedPage callback", () => {
        const mockSetSelectedPage = jest.fn();
        const props: PaginationProps = {
            slots: [1, 2, 3],
            currentPageNum: 2,
            setSelectedPage: mockSetSelectedPage,
        };
        renderWithBase(<Pagination {...props} />);

        fireEvent.click(screen.getByText("Previous"));
        expect(mockSetSelectedPage).toHaveBeenLastCalledWith(1);

        fireEvent.click(screen.getByText("1"));
        expect(mockSetSelectedPage).toHaveBeenLastCalledWith(1);

        fireEvent.click(screen.getByText("2"));
        expect(mockSetSelectedPage).toHaveBeenLastCalledWith(2);

        fireEvent.click(screen.getByText("3"));
        expect(mockSetSelectedPage).toHaveBeenLastCalledWith(3);

        fireEvent.click(screen.getByText("Next"));
        expect(mockSetSelectedPage).toHaveBeenLastCalledWith(3);
    });

    test("Renders the expected markup when the current page is between two other pages", () => {
        const props: PaginationProps = {
            slots: [1, OVERFLOW_INDICATOR, 6, 7, 8, 9, OVERFLOW_INDICATOR],
            currentPageNum: 7,
            setSelectedPage: jest.fn(),
        };
        const { asFragment } = renderWithBase(<Pagination {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("Renders the expected markup when the current page is the last page", () => {
        const props: PaginationProps = {
            slots: [1, 2],
            currentPageNum: 2,
            setSelectedPage: jest.fn(),
        };
        const { asFragment } = renderWithBase(<Pagination {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
