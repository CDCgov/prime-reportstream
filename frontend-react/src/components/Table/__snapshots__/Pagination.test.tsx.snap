// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > Renders the expected markup when the current page is between two other pages 1`] = `
<DocumentFragment>
  <nav
    aria-label="Pagination"
    class="usa-pagination"
    data-testid="Pagination"
  >
    <ul
      class="usa-pagination__list"
    >
      <li
        class="usa-pagination__item usa-pagination__arrow"
      >
        <button
          aria-label="Previous page"
          class="usa-button usa-button--unstyled rs-pagination-arrow-button usa-pagination__previous-page"
          data-testid="Previous page"
          type="button"
        >
          <svg
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="usa-pagination__link-text"
          >
            Previous
          </span>
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-label="Page 1"
          class="usa-button usa-button--unstyled usa-pagination__button"
          data-testid="Page 1"
          type="button"
        >
          1
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__overflow"
        role="listitem presentation"
      >
        <span>
          …
        </span>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-label="Page 6"
          class="usa-button usa-button--unstyled usa-pagination__button"
          data-testid="Page 6"
          type="button"
        >
          6
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-current="page"
          aria-label="Page 7"
          class="usa-button usa-button--unstyled usa-pagination__button usa-current"
          data-testid="Page 7"
          type="button"
        >
          7
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-label="Page 8"
          class="usa-button usa-button--unstyled usa-pagination__button"
          data-testid="Page 8"
          type="button"
        >
          8
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-label="Page 9"
          class="usa-button usa-button--unstyled usa-pagination__button"
          data-testid="Page 9"
          type="button"
        >
          9
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__overflow"
        role="listitem presentation"
      >
        <span>
          …
        </span>
      </li>
      <li
        class="usa-pagination__item usa-pagination__arrow"
      >
        <button
          aria-label="Next page"
          class="usa-button usa-button--unstyled rs-pagination-arrow-button usa-pagination__next-page"
          data-testid="Next page"
          type="button"
        >
          <span
            class="usa-pagination__link-text"
          >
            Next
          </span>
          <svg
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination > Renders the expected markup when the current page is the last page 1`] = `
<DocumentFragment>
  <nav
    aria-label="Pagination"
    class="usa-pagination"
    data-testid="Pagination"
  >
    <ul
      class="usa-pagination__list"
    >
      <li
        class="usa-pagination__item usa-pagination__arrow"
      >
        <button
          aria-label="Previous page"
          class="usa-button usa-button--unstyled rs-pagination-arrow-button usa-pagination__previous-page"
          data-testid="Previous page"
          type="button"
        >
          <svg
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="usa-pagination__link-text"
          >
            Previous
          </span>
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-label="Page 1"
          class="usa-button usa-button--unstyled usa-pagination__button"
          data-testid="Page 1"
          type="button"
        >
          1
        </button>
      </li>
      <li
        class="usa-pagination__item usa-pagination__page-no"
      >
        <button
          aria-current="page"
          aria-label="last page, Page 2"
          class="usa-button usa-button--unstyled usa-pagination__button usa-current"
          data-testid="last page, Page 2"
          type="button"
        >
          2
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
